### PR TITLE
#12246: Remove `perf_microbenchmark/common/work_split.hpp`

### LINKED ISSUE #12246: Remove file `tests/tt_metal/tt_metal/perf_microbenchmark/common/work_split.hpp`
Master Issue : #12204
PR : https://github.com/tenstorrent/tt-metal/pull/12247

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
Issue : #12246 , Master Issue : #12204
Work Done : Removed `tests/tt_metal/tt_metal/perf_microbenchmark/common/work_split.hpp`
[CI](https://github.com/tenstorrent/tt-metal/actions/runs/10700190078) - Passed

CI :

- [All post-commit tests](https://github.com/tenstorrent/tt-metal/actions/runs/10714514828) - PASSED
- [[post-commit] C++ tests](https://github.com/tenstorrent/tt-metal/actions/runs/10714515568) - PASSED
- [(Single-card) Device perf regressions](https://github.com/tenstorrent/tt-metal/actions/runs/10714516389) - PASSED
- [(Single-card) Model perf tests](https://github.com/tenstorrent/tt-metal/actions/runs/10714517315) - PASSED
- [(T3K) T3000 model perf tests](https://github.com/tenstorrent/tt-metal/actions/runs/10714518131) - PASSED as in main
- [(T3K) T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/runs/10714518972) - PASSED
- [(T3K) T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/runs/10714519640) -
- [Nightly fast dispatch tests](https://github.com/tenstorrent/tt-metal/actions/runs/10714520487) - 
- [(T3K) T3000 frequent tests](https://github.com/tenstorrent/tt-metal/actions/runs/10714520487) -
- [(Single-card) Demo tests](https://github.com/tenstorrent/tt-metal/actions/runs/10714521914) - 
- [(T3K) T3000 profiler tests](https://github.com/tenstorrent/tt-metal/actions/runs/10714522690) - PASSED
- [metal - Run microbenchmarks](https://github.com/tenstorrent/tt-metal/actions/runs/10750542823) - PASSED

### CODE REVIEW

**Review by Aswinmcw — APPROVED** [review]
LGTM

**Review by tt-aho — APPROVED** [review]
lgtm.
@VirdhatchaniKN you should review what tests/pipelines are actually affected by changes you're making and run the appropriate pipeline.
Ex. the specific pipeline that should have been run for this change was https://github.com/tenstorrent/tt-metal/actions/workflows/metal-run-microbenchmarks.yaml, and not most of the other ones (ex T3K)

**Review by yugaoTT — APPROVED** [review]

**Comment by bbradelTT** [comment]
Why does this look almost identical to https://github.com/tenstorrent/tt-metal/pull/11834 ?

File: ttnn/ttnn/core.py [hunk 1]
```diff
@@ -44,6 +44,36 @@ def is_sharded(tensor) -> bool:
...
+    Returns a CoreRangeSet from number of cores
+    """
+    return ttnn._ttnn.operations.core.num_cores_to_core_range_set(
+        target_num_cores,
+        grid_size,
+        row_wise,
+    )
+
+
+def num_cores_to_corerange_set(
```
Comment (bbradelTT) [inline ttnn/ttnn/core.py:62]: Same problem as in https://github.com/tenstorrent/tt-metal/pull/11834/files

**Comment by VirdhatchaniKN** [comment]
> Why does this look almost identical to #11834 ?

Yes.... As this file that we have removed was not listed in the migration task , we put it as a separate PR to avoid conflicts.

**Review by bbradelTT — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
- https://github.com/tenstorrent/tt-metal/actions/runs/10750527660 - passed
- https://github.com/tenstorrent/tt-metal/actions/runs/10750528573 - passed
- https://github.com/tenstorrent/tt-metal/actions/runs/10750538905
- https://github.com/tenstorrent/tt-metal/actions/runs/10750557223 - passed

**Comment by VirdhatchaniKN** [comment]
- https://github.com/tenstorrent/tt-metal/actions/runs/10750540060
- https://github.com/tenstorrent/tt-metal/actions/runs/10750537488
- https://github.com/tenstorrent/tt-metal/actions/runs/10750542438 - passed
- https://github.com/tenstorrent/tt-metal/actions/runs/10750541181/job/29819600221

**Comment by VirdhatchaniKN** [comment]
- https://github.com/tenstorrent/tt-metal/actions/runs/10750529412 - passed
- https://github.com/tenstorrent/tt-metal/actions/runs/10750530207 - passed
- https://github.com/tenstorrent/tt-metal/actions/runs/10750542823 - passed
- https://github.com/tenstorrent/tt-metal/actions/runs/10750541943 - PASSED AS IN MAIN

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/10758057871 - passed

**Pull request merged**